zerver/logging_handlers.py

```diff
@@ -46,18 +46,18 @@ def add_request_metadata(report: Dict[str, Any], request: HttpRequest) -> None:
         from django.contrib.auth.models import AnonymousUser
         user_profile = request.user
         if isinstance(user_profile, AnonymousUser):
-            user_full_name = None
-            user_email = None
+            user_realm = None
+            user_id = None
```

Comment:
I think changing this requires also changing `zerver/lib/error_notify.py`, the code that actually formats these.

You should probably test-deploy this in a test production environment (or figure out how to configure these to be sent and go to /emails in development) for testing; this code is relatively fragile.

Reply:
Actually I'm probably most excited about making a convenient way to test this manually in development (we can potentially even have a `dev_urls.py` API endpoint whose sole purpose is to throw a 500).  There's probably 10 other things we'll want to do to improve those emails, not just this, and having it be convenient to test that would make it way more doable